src/lambdify.jl

```diff
@@ -105,9 +105,13 @@ end
 
 _arguments(x::SymbolicObject) = collect(args(x))
 
-function _similarterm(t::SymbolicObject, f, args, symtype=nothing;
-                      metadata=nothing, exprhead=:call)
-    f(args...) # default
+function _similarterm(T::Type{<:SymbolicObject}, head::Symbol, args, metadata)
+    if head == :call
+        f, as... = args
+        return f(as...)
+    else
+        @show :head
```

Comment:
?

Reply:
Yes, this was a mess. I think I better understand what to do here now after reading what Symbolics did (and not what is being done for Expr in TermInterface.